feat: use and customise Carbon light theme

eddybrando opened the pull request:
This PR refactors the current Carbon theme in order to extend the **white** and not the **gray 100** version.

Some other improvements were made to the way we import CSS styles and variables to avoid, for instance, repeating the import of the _copy_ styles files in various components.

The special styling for the "metal" page has algo been simplified.

As part of the refactory, we're also now using the appropriate Carbon color tokens when possible.

---

Closes #965 

Addresses the FOUT part of #1569

y4izus commented:
@eddybrando do you think you could resolve this issue #1366 while doing this refactor?

y4izus commented on nuxt.config.ts:
```diff
@@ -131,7 +124,8 @@ const config: NuxtConfig = {
       ignoreNotFoundWarnings: true
-    }]
+    }],
+    '@nuxtjs/style-resources',
```
Per offline conversations, we move the `@nuxtjs/stype-resources` into `buildModules` to follow the [instructions on the setup](https://github.com/nuxt-community/style-resources-module/#setup)

eddybrando replied:
@delapuente FYI

y4izus reviewed (approved):
Great refactor!! this is very useful!! thank you very much Eddy!

y4izus commented:
I noticed that some styles are broken, could you take a look at them, please?
- The hover effect on the "copy" button doesn't work
<img width="666" alt="Captura de pantalla 2021-04-20 a las 9 42 59" src="https://user-images.githubusercontent.com/17231966/115357309-cf5f8780-a1bc-11eb-851a-befb2a0366ec.png">

- When selecting an option on the quick start, it turns blue
<img width="675" alt="Captura de pantalla 2021-04-20 a las 9 44 38" src="https://user-images.githubusercontent.com/17231966/115357551-0a61bb00-a1bd-11eb-8295-6549dc2d4ac3.png">

- Also on the quick start, when clicking on the accordion, the blue appears:
<img width="671" alt="Captura de pantalla 2021-04-20 a las 9 45 33" src="https://user-images.githubusercontent.com/17231966/115357661-2a917a00-a1bd-11eb-8013-4bb1f78f05ce.png">

- The menu link is not purple when it's active because we are in that page. On mobile menu it works
<img width="1108" alt="Captura de pantalla 2021-04-20 a las 9 46 46" src="https://user-images.githubusercontent.com/17231966/115357844-557bce00-a1bd-11eb-9fc8-d596ec287071.png">

- On the overview page, when clicking on the accordion, again the blue:
<img width="607" alt="Captura de pantalla 2021-04-20 a las 9 48 19" src="https://user-images.githubusercontent.com/17231966/115358102-9c69c380-a1bd-11eb-8bc9-da281771f525.png">

- Checkboxes has not the proper styles (refactor vs production)(learn page):
<img width="233" alt="Captura de pantalla 2021-04-20 a las 9 51 33" src="https://user-images.githubusercontent.com/17231966/115358606-1ef28300-a1be-11eb-80f1-d2a4a2625fb2.png">
<img width="258" alt="Captura de pantalla 2021-04-20 a las 9 52 41" src="https://user-images.githubusercontent.com/17231966/115358651-29ad1800-a1be-11eb-9e9c-05c91c0bb370.png">

- Blue instead of purple (learn page):
<img width="417" alt="Captura de pantalla 2021-04-20 a las 9 53 09" src="https://user-images.githubusercontent.com/17231966/115358748-40ec0580-a1be-11eb-99c2-64919d0a407c.png">

- Blue instead of purple (learn page):
<img width="470" alt="Captura de pantalla 2021-04-20 a las 9 53 50" src="https://user-images.githubusercontent.com/17231966/115358820-56f9c600-a1be-11eb-800f-df257a8b58aa.png">

- Blue instead of purple (events page) (it's the same for all the filters):
<img width="493" alt="Captura de pantalla 2021-04-20 a las 9 55 11" src="https://user-images.githubusercontent.com/17231966/115359014-87d9fb00-a1be-11eb-81d4-87e6ddb9e5d8.png">

- Blue instead of purple (events page) (refactor vs production)
<img width="170" alt="Captura de pantalla 2021-04-20 a las 9 57 19" src="https://user-images.githubusercontent.com/17231966/115359264-cec7f080-a1be-11eb-922b-f47e004b1cb9.png">
<img width="168" alt="Captura de pantalla 2021-04-20 a las 9 57 48" src="https://user-images.githubusercontent.com/17231966/115359347-e0a99380-a1be-11eb-8ba8-f3e7b3ab7044.png">

- Blue instead of purple  (events page) (refactor vs production)
<img width="420" alt="Captura de pantalla 2021-04-20 a las 9 58 23" src="https://user-images.githubusercontent.com/17231966/115359509-0898f700-a1bf-11eb-875f-db7ce761eb38.png">
<img width="425" alt="Captura de pantalla 2021-04-20 a las 9 58 40" src="https://user-images.githubusercontent.com/17231966/115359519-0b93e780-a1bf-11eb-9a18-aeac27c7a0c4.png">

- Same on the advocate's page for the filters

eddybrando commented:
@y4izus great catches! Thanks a lot for the detailed review. I just checked and fixed all of the mentioned bugs and fixed #1366 as well.

As we discussed offline and you pointed out, this is a prime example of our need for an automated visual regression testing process, that could help us spare a lot of time in this kind of reviews and avoid regressions.

y4izus reviewed (approved):
Now it's perfect Eddy!! thank you very much for the efforts!
